Write 1 in the ts-numbers when series.width == 1

flomnes opened the pull request:
Before that patch, random numbers were drawn, ignored in the simulation, but still written in ts-numbers.

flomnes committed:
Write 1 in the ts-numbers when series.width == 1

Before that patch, random numbers were drawn, ignored in the
simulation, but still written in ts-numbers.

flomnes committed:
Remove useless conversion to long

guilpier-code commented on src/solver/aleatoire/alea_tirage_au_sort_chroniques.cpp:
```diff
@@ -65,33 +65,44 @@ static void InitializeTimeSeriesNumbers_And_ThermalClusterProductionCost(
             assert(year < data.timeseriesNumbers.height);
-            ptchro.Consommation
-              = (data.series.width != 1) ? (long)data.timeseriesNumbers[0][year] : 0; // zero-based
+            if (data.series.width == 1)
```
As **timeseriesNumbers** was set previously in **timeseries-numbers.cpp**, it should already be set to **0** if there is only one available time series for some given area and year.
Said in another way, there should be a need for setting **timeseriesNumbers** here.

flomnes replied:
What you mean is that `timeseriesNumbers` should be set to 0 if `width==1` in **timeseries-numbers.cpp**. Then we would introduce a dependency to timeseries in **timeseries-numbers.cpp**, I'm not sure we want to do that.

flomnes replied:
Actually, that dependency already exists in **timeseries-numbers.cpp**. Therefore, the TS numbers provided by `TimeSeriesNumbers::Generate` are wrong when `width==1`. I agree : the fix should be applied there, not here.

flomnes committed:
Revert "Remove useless conversion to long"

This reverts commit c43ab0b07741dff4e94d00bdc9dca5c9a58aff53.

flomnes committed:
Revert "Write 1 in the ts-numbers when series.width == 1"

This reverts commit 8d32e2a3e08c27f93fea0df6c01beae0b034e469.

flomnes committed:
Remove useless .entry

use operator[] instead

flomnes committed:
Set timeseriesNumbers to 0 if width==1

flomnes committed:
[skip ci] Spacing

flomnes committed:
Remove failed tests

When intramodal is set to true, it can't be expected that all
time-series numbers be equal, even if width==1 for some clusters.

flomnes committed:
[sonar] Fix sonar code smells

guilpier-code commented:
As you now set **timeseriesNumbers** to **0** when there is only **1** TS for given year, area, ... in **timeseries-numbers.cpp**, file **alea_tirage_au_sort_chroniques.cpp** can be simplified. 
Example : 
```c++
ptchro.Consommation = data.timeseriesNumbers[0][year];
```

flomnes committed:
PR remarks

guilpier-code commented on src/tests/src/solver/simulation/tests-ts-numbers.cpp:
```diff
@@ -354,7 +353,6 @@ BOOST_AUTO_TEST_CASE(two_areas_3_renew_clusters_with_respectively_4_4_1_ready_ma
 	// TS number checks
 	uint year = 0;
 	BOOST_CHECK_EQUAL(rnCluster_12->series->timeseriesNumbers[0][year], rnCluster_11->series->timeseriesNumbers[0][year]);
-	BOOST_CHECK_EQUAL(rnCluster_21->series->timeseriesNumbers[0][year], rnCluster_11->series->timeseriesNumbers[0][year]);
```
ok, so before the patch, **timeseriesNumbers** is to something != 0 even if the TS actual size is **1**.
The patch in this test makes **rnCluster_21** useless. And we should rename that test.
And maybe make another test that show te new behavior.

flomnes replied:
I will just remove these obsolete clusters and rename the corresponding tests. My fix does not change the behavior of `Antares::Solver::TimeSeriesNumbers::Generate`.

The correct way to fix the bug would be to change `Antares::Solver::TimeSeriesNumbers::Generate`, but unfortunately it is called before the timeseries generator (the one that generates the coefficients of the TS, not the numbers). A large refactor would be necessary to include that change in `TimeSeriesNumbers::Generate`.

guilpier-code commented on src/tests/src/solver/simulation/tests-ts-numbers.cpp:
```diff
@@ -233,7 +233,6 @@ BOOST_AUTO_TEST_CASE(two_areas_3_thermal_clusters_with_respectively_4_1_4_ready_
 
 	// TS number checks
 	uint year = 0;
-	BOOST_CHECK_EQUAL(thCluster_12->series->timeseriesNumbers[0][year], thCluster_11->series->timeseriesNumbers[0][year]);
```
Same remark as below (for renewable clusters)

guilpier-code replied:
bla

flomnes committed:
Revert "PR remarks"

This reverts commit b4abb71ef92a91d16476d8b329e387d14358618f.

flomnes committed:
Store ts-numbers at the end of the simulation

flomnes committed:
[skip ci] Add explanatory comment

flomnes committed:
PR remarks

flomnes committed:
[skip ci] Use std::for_each, code formatting
